include/swift/AST/ASTContext.h

```diff
@@ -869,6 +870,21 @@ class ASTContext final {
   /// This guarantees that resulted \p names doesn't have duplicated names.
   void getVisibleTopLevelModuleNames(SmallVectorImpl<Identifier> &names) const;
 
+  struct OverrideSignatureKey {
+    GenericSignature *baseMethodSig;
+    GenericSignature *derivedClassSig;
+    Type superclassTy;
+
+    OverrideSignatureKey(GenericSignature *baseMethodSignature,
+                         GenericSignature *derivedClassSignature,
+                         Type superclassType)
+        : baseMethodSig(baseMethodSignature),
+          derivedClassSig(derivedClassSignature), superclassTy(superclassType) {
+    }
+  };
+
+  llvm::DenseMap<OverrideSignatureKey, GenericSignature *> overrideSigCache;
```

Comment:
This should be a field of ASTContext::Impl, not of ASTContext itself. ASTContext itself just holds a pointer to the Impl.